linker/Tests/Mono.Linker.Tests.Cases.Expectations/Metadata/CompileAssemblyBeforeAttribute.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace Mono.Linker.Tests.Cases.Expectations.Metadata {
+	/// <summary>
+	/// Use to compile an assembly before compiling the main test case executabe
+	/// </summary>
+	[AttributeUsage (AttributeTargets.Class, AllowMultiple = true)]
+	public class CompileAssemblyBeforeAttribute : BaseMetadataAttribute {
```

Comment:
Would much prefer name which says this is setup phase for the test. Perhaps `class SetupCompileBefore` ?